test/model_factory_test.py

```diff
@@ -168,3 +168,77 @@ def test_from_dict_fails_on_missing_cfg_types(self):
             ConfigFactory.from_dict({})
 
 
+class ConfigFactoryUnitTest(unittest.TestCase):
+    def setUp(self):
+        types = {
+            'a_type':
+            {
+                'model': { 'cfg_type_name': 'a_type', 'type': 'NamedModelElement' }
+            },
+            'cfg_set':
+            {
+                'model': { 'cfg_type_name': 'cfg_set', 'type': 'ConfigurationSet' }
+            }
+        }
+        # config sets
+        cfg_sets = {
+                'first_set': {'a_type': 'first_value_of_a'},
+                'second_set':
+                {
+                    'a_type':
+                    {
+                        'config_names':
+                        {
+                            'second_value_of_a',
+                            'third_value_of_a'
+                        }
+                    }
+                }
+        }
+        # value definitions
+        values = {
+                'first_value_of_a': {'some_value': 123},
+                'second_value_of_a': {'some_value': 42},
+                'third_value_of_a': {'some_value': 'xxx'},
+        }
+        self.raw_dict = {'cfg_types': types, 'cfg_set': cfg_sets, 'a_type': values}
+
+    def test_iterate_config_names_handles_lists(self):
+        factory = ConfigFactory.from_dict(self.raw_dict)
+        cfg_list = list(
+            factory.iter_config_names(
+                cfg_set_name='second_set',
+                cfg_type_name='a_type',
+            )
+        )
+        self.assertEqual(len(cfg_list), 2)
```

Comment:
self.assertEqual(cfg_list, [<first>, <second>]
better yet: use sets, as we do not guarantee the order